sympy/printing/maple.py

```diff
@@ -0,0 +1,216 @@
+}
+
+spec_relational_ops = {
+    # Sympy -> Maple
+    '==': '=',
+    '!=': '<>'
+}
+
+
+class MapleCodePrinter(CodePrinter):
+    """
+    Printer which converts single sympy expressions into single.
+    """
+    printmethod = "_maple"
+    language = "maple"
+
+    def __init__(self, settings=None):
+        if settings is None:
+            settings = dict()
+        super(MapleCodePrinter, self).__init__(settings)
+        self.known_functions = dict(known_functions)
+        userfuncs = settings.get('user_functions', {})
+        self.known_functions.update(userfuncs)
+
+    def _get_statement(self, codestring):
+        return "%s;" % codestring
+
+    def _get_comment(self, text):
+        return "# {0}".format(text)
+
+    def _declare_number_const(self, name, value):
+        return "{0} := {1};".format(name,
+                                    value.evalf(self._settings['precision']))
+
+    def _format_code(self, lines):
+        return lines
+
+    def _print_Assignment(self, expr):
+        lhs = expr.lhs
+        rhs = expr.rhs
+        return "{lhs} := {rhs}".format(lhs=lhs, rhs=rhs)
+
+    def _print_Pow(self, expr):
+        PREC = precedence(expr)
+        if expr.exp == -1:
+            return '1/%s' % (self.parenthesize(expr.base, PREC))
+        elif expr.exp == 0.5 or expr.exp == S(1) / 2:
+            return 'sqrt(%s)' % self._print(expr.base)
+        elif expr.exp == -0.5 or expr.exp == -S(1) / 2:
+            return '1/sqrt(%s)' % self._print(expr.base)
+        else:
+            return '%s^%s' % (
+                self.parenthesize(expr.base, PREC),
+                self.parenthesize(expr.exp, PREC))
+
+    def _print_Rational(self, expr):
+        p, q = int(expr.p), int(expr.q)
+        return '%d/%d' % (p, q)
+
+    def _print_Relational(self, expr):
+        lhs_code = self._print(expr.lhs)
+        rhs_code = self._print(expr.rhs)
+        op = expr.rel_op
+        if op in spec_relational_ops:
+            op = spec_relational_ops[op]
+        return "{0} {1} {2}".format(lhs_code, op, rhs_code)
+
+    def _print_NumberSymbol(self, expr):
+        return number_symbols[expr]
+
+    def _print_NegativeInfinity(self, expr):
+        return '-infinity'
+
+    def _print_Infinity(self, expr):
+        return 'infinity'
+
+    def _print_Idx(self, expr):
+        return self._print(expr.label)
+
+    def _print_BooleanTrue(self, expr):
+        return "true"
+
+    def _print_BooleanFalse(self, expr):
+        return "false"
+
+    def _print_bool(self, expr):
+        return str(expr).lower()
+
+    def _print_NaN(self, expr):
+        return 'undefined'
+
+    def _get_matrix(self, expr, sparse=False):
+        if expr.cols == 0 or expr.rows == 0:
+            _strM = 'Matrix([], storage = {storage}})'.format(
+                storage='sparse' if sparse else 'rectangular')
+        else:
+            _strM = 'Matrix({list}, storage = {storage})'.format(
+                list=str(expr.tolist()),
```

Comment:
I don't think that it is a good idea to call `str`, than calling `self._print`
because `str` will call StrPrinter which is designed to print things for displays, not some valid code.
So the expression printed can be wrong, if it is not as simple as a number or a single symbol.

Reply:
I think, the type of `M.tolist()` is a list. So `str(M.tolist())` should call __str__ defined in list class.

Reply:
`__str__` defined in `list` is less likely to be changed in further development.

Reply:
Here is the example having difference

```
>>> x, y = symbols('x y')
>>> maple_code(Matrix([[S.Exp1]]))
'Matrix([[E]], storage = rectangular)'
>>> maple_code([[S.Exp1]])
'[[exp(1)]]'
```

As you can see, if you use `str`, it doesn't recurse with maple code printer's printing methods to print any content inside the list, but blindly to an another nonsensical expression.

Reply:
Ok, fine, I will correct it.